### PR TITLE
catalog implementation limitations in documentation

### PR DESCRIPTION
**Pull request by dhalbert**
I made a sweep through the implementations and came up with this list of limitations. Some are "to be implemented", but many are inherent. Documenting them will save support time.

### CODE REVIEW

File: shared-bindings/analogio/AnalogOut.c [hunk 1]
```diff
@@ -50,7 +50,11 @@
 //|     def __init__(self, pin: microcontroller.Pin) -> None:
 //|         """Use the AnalogOut on the given pin.
 //|
-//|         :param ~microcontroller.Pin pin: the pin to output to"""
+//|         :param ~microcontroller.Pin pin: the pin to output to
+//|
+//|         **Limitations:** Not available on nRF, RP2040, Spresense: there is no on-chip DAC.
+//|          Espressif: avaiable only on ESP32 and ESP32-S2; other chips do not have a DAC.
```
Comment (kattni) [inline shared-bindings/analogio/AnalogOut.c:56]:
```suggestion
//|          Espressif: available only on ESP32 and ESP32-S2; other chips do not have a DAC.
```

**Review by kattni — CHANGES_REQUESTED** [review]
This is excellent! Caught one spelling error. See below.

File: shared-bindings/analogio/AnalogOut.c (hunk 1 above)
Comment (dhalbert) [reply shared-bindings/analogio/AnalogOut.c]: Fixed! Thanks for spotting that.

**Review by kattni — APPROVED** [review]
Thanks for adding this!

**Pull request merged**